### PR TITLE
Feature/20 Registration Service API ADR

### PR DESCRIPTION
**Pull request by Izzzu**
## What this PR changes/adds

This PR adds retroactively decision record documentation about Registration Service API.  

## Why it does that

To keep track of the development decisions.

## Further notes

More abstract documentation containing high level flows will be added to MVD repo.

## Linked Issue(s)

Closes https://github.com/agera-edc/MinimumViableDataspace/issues/20

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [x] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/RegistrationService/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/RegistrationService/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: docs/developer/decision-records/2022-06-13-registration-service-api/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+### Add participant
+
+Calling POST method to add participant triggers an enrollment process of a new Dataspace member. The new participant is saved to the store for further 
+processing. 
+
+### List participants
+
+Calling GET method to list participants returns a list of all participants that are saved in the store.
+
```
Comment (bscholtes1A) [inline docs/developer/decision-records/2022-06-13-registration-service-api/README.md:48]: Not part of this PR but would be great to also have `update` and `delete` operations.

File: docs/developer/decision-records/2022-06-13-registration-service-api/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Registration Service API
+
+Registration Service is a component of a Dataspace Authority. 
+One of the responsibilities of the Registration Service is the management of the participants in the Dataspace.
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2022-06-13-registration-service-api/README.md:4]: I'd say that that is it's _only_ responsibility. If not, I'd list the other responsibilities.

File: docs/developer/decision-records/2022-06-13-registration-service-api/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+The Registration Service is written in Java and uses the runtime framework and modules from [EDC](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector).
+
+## OpenApi 
+
+The Registration Service API definition is available in an [OpenApi yaml file](../../../../resources/openapi/yaml/registration-service.yaml). 
+
+## API Rest client
+
+The Registration Service API can be called using the [rest-client](../../../../rest-client) generated from OpenApi definition.
```
Comment (paullatzelsperger) [inline docs/developer/decision-records/2022-06-13-registration-service-api/README.md:18]: Not sure we really want to include the Java client code in the "server" code. If anything, we should publish this as separate Maven artifact down the line.

Comment (Izzzu) [reply docs/developer/decision-records/2022-06-13-registration-service-api/README.md:18]: It's mentioned below that the Registration Service is currently not published to any artifactory so the client can be used for local development. Would you change/add something to that? Or are you suggesting to remove `API Rest Client` section from the ADR?

File: docs/developer/decision-records/2022-06-13-registration-service-api/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+```java
+ApiClient apiClient = ApiClientFactory.createApiClient(API_URL);
+RegistryApi api = new RegistryApi(apiClient);
+```
+
+Currently the Registration Service is not published to any public artifactory so the rest client can be used in local development after publishing the
```
Comment (Izzzu) [inline docs/developer/decision-records/2022-06-13-registration-service-api/README.md:25]:
```suggestion
Currently the Registration Service client is not published to any public artifactory so the rest client can be used in local development after publishing the 
```

File: docs/developer/decision-records/2022-06-13-registration-service-api/README.md (hunk 3 above)
Comment (paullatzelsperger) [reply docs/developer/decision-records/2022-06-13-registration-service-api/README.md:18]: I'm doubtful that there is much value in shipping REST client code directly with the server code. 
It's OK to leave it in for now, for development/testing purposes, but eventually the client code should not be shipped with the server code.
